src/GraphQL/Internal/Validation.hs

```diff
@@ -386,12 +472,77 @@ resolveFragmentDefinitions allFragments =
           modify (Set.insert name)
           FragmentSpread name directives <$> resolveFragment' definition
 
+-- * Variables
+
+data VariableDefinition = VariableDefinition Variable AST.Type (Maybe Value) deriving (Eq, Ord, Show)
```

Comment:
Does `Maybe Value` mean the type is nullable? BTW it might be worth having our own `Nullable` instead of `Maybe` because of all the problems that `Maybe` has caused in aeson? The name would also better communicate what we use it for.